react/rules/BRA.js

```diff
@@ -123,50 +118,77 @@ export default {
       name: 'corporateName',
       maxLength: 100,
       label: 'corporateName',
+      required: true,
     },
     {
       name: 'corporateDocument',
       maxLength: 30,
       label: 'BRA_cnpj',
+      required: true,
       mask: value => msk.fit(value, '99.999.999/9999-99'),
-      validate: value => {
-        const cleanValue = value.replace(/[^\d]/g, '')
+      validate: cnpj => {
+        if (!cnpj) {
+          return false
+        }
 
-        if (cleanValue.length !== 14) return false
+        cnpj = cnpj.replace(/[^\d]+/g, '')
 
-        const isRepeatedNum = '0123456789'
-          .split('')
-          .some(digit => digit.repeat(14) === cleanValue)
+        if (cnpj.length !== 14) {
+          return false
+        }
 
-        if (isRepeatedNum) return false
+        if (
+          cnpj === '00000000000000' ||
+          cnpj === '11111111111111' ||
+          cnpj === '22222222222222' ||
+          cnpj === '33333333333333' ||
+          cnpj === '44444444444444' ||
+          cnpj === '55555555555555' ||
+          cnpj === '66666666666666' ||
+          cnpj === '77777777777777' ||
+          cnpj === '88888888888888' ||
+          cnpj === '99999999999999'
+        ) {
+          return false
+        }
 
-        const firstWeights = '543298765432'.split('')
-        const firstReduce = cleanValue
-          .split('')
-          .slice(0, 12)
-          .reduce(
-            (acc, cur, index) =>
-              acc + parseInt(cur, 10) * parseInt(firstWeights[index], 10),
-            0
-          )
+        // Validate DVs
+        let size = cnpj.length - 2
+        let numbers = cnpj.substring(0, size)
+        const digitos = cnpj.substring(size)
```

Comment:
:eyes:

Reply:
This is the validation that inStore currently uses. I tested the profile-form validation and it didn't work on some CNPJs I tested

Reply:
`digitos`.....

Reply:
Good eyes